### PR TITLE
WINTERMUTE: Solve 'Chinese' language conflicts. Add more game entries

### PR DESCRIPTION
**Pull request by aviloria**
It seems that chinese language for reversion2 was not updated in November 2019

### CODE REVIEW

**Comment by lolbot-iichan** [comment]
@aviloria @somaen @tobiatesan 
Reversion 2 for Linux detection & processing causes some troubles I want to discuss.
Review for adding Linux version was already approved, so let's discuss it here please?

"August 2019" version of Reversion 2 for Linux has a single Chinese translation in `xlanguage_nz.dcp` ("Chinese (China)") package.
"November 2019" version of Reversion 2 for Linux has three Chinese translations: two of them are actually used and stored at `xlanguage_zh_s.dcp` ("Chinese (China)") and `xlanguage_zh_t.dcp` ("Chinese (Taiwan)"), third is unmodified file of "August 2019" version that seems to be actually unused (as for Windows version - I can't select it from game launcher). So, current `detection_table.h` produces 3 detects for "Chinese (China)" for "November 2019" version of Reversion 2 for Linux:
* Reversion: The Meeting (Steam, August 2019) (Chinese)
* Reversion: The Meeting (Steam, November 2019) (Chinese)
* Reversion: The Meeting (Steam, November 2019) (Simplified Chinese)

"Reversion: The Meeting (Steam, November 2019) (Chinese)" is duplicate that should be removed.

As for "Chinese" / "Simplified Chinese" detection - this is highly debatable. 
A. From one point of view - game HAS two different translations, so why don't we give user choice to play August or November version of translation since he is lucky to have both.
B. From another point of view - old translation seems to be disabled in actual game, so why should we emulate something that is not available without ScummVM.

If we choose to show both Chinese versions, we should decide on a way to do this, from user point of view and from `BaseFileManager::registerPackages()` point of view. 
A1. One way is to introduce new `Common::ZH_ANY` language, so that `xlanguage_nz.dcp` would be `Common::ZH_ANY`, `xlanguage_zh_s.dcp` would be `Common::ZH_CNA` and `xlanguage_zh_t.dcp` would be `Common::ZH_TWN`. This seems a valid option, since we have `Common::EN_ANY` already, but since we don't have "when only one game version exist" edition of Chinese, there maybe was a reason for it.
A2. Another way is to introduce a new flag at `enum WintermuteGameFeatures`, so that `BaseFileManager::registerPackages()` would check this flag and ignore `xlanguage_nz.dcp` and/or `xlanguage_zh_s.dcp` according to current lang and flag. This looks ugly, but the same code could be easily reused to do stuff like loading only SD/HD packages for games that ships a lot of packages alltogether (Hello, Shadows of Vatican).
A3. There are other ways as well...

B1. If we choose to show only one Chinese version, `BaseFileManager::registerPackages()` should be fixed to ignore `xlanguage_nz.dcp` if package `xlanguage_zh_s.dcp` exists.

**Comment by aviloria** [comment]
@lolbot-iichan: I already removed the following 2 entries:
- Linux: Reversion: The Meeting (Steam, November 2019) (Chinese) 
- Mac: Reversion: The Meeting (Steam, November 2019) (Chinese)
As you have said they are the same as the August 2019 releases

I also notice that Windows version includes 3 different Chinese flavours: nz, zh_s and zh_t
The first one cannot be selected in the game menu, it has not been updated since November 2016 and seem redundant with the other 2. 
I agree to remove that entry from the Windows package too.

**Comment by lolbot-iichan** [comment]
> @lolbot-iichan: I already removed the following 2 entries:
Oops. Sorry, updated my comment.

>  I agree to remove that entry from the Windows package too.
It's not only about removing. We should either provide a detect for 3 Chinese versions every time when it's possible, or never provide 2 detections for `Chinese(China)`.

We can't remove "Reversion: The Meeting (Steam, August 2019) (Chinese)" because it's required to detect Chinese August 2019, but in current state it fires on November 2019 as well, since it has the same `data.dcp`, `Linux.dcp` & `xlanguage_nz.dcp`, which makes November 2019 having 3 detects.

**Comment by aviloria** [comment]
As far as I know, Linux and Mac versions look good right now (removed the Nov2019 chinese entries for being redundant)

Aug2019 and Nov2019 Windows versions contains the 3 chinese flavours... but as I have said, one of them (nz) has not been updated since Nov2016, so the easiest solution is to remove that entries from Windows Aug2019 and Nov2019

**Comment by aviloria** [comment]
I have commented out the 2 conflicting entries until you decide how to proceed.

**Comment by lolbot-iichan** [comment]
>As far as I know, Linux and Mac versions look good right now (removed the Nov2019 chinese entries for being redundant)

I have tried building your branch and I see 2 detections of Chinese(China) for November 2019 Linux edition of Reversion2:

изображение

I download it with `download_depot 281060 281062 3573144182407273735`.

Detects are "// Reversion: The Meeting (Steam, August 2019) (Chinese)" and "// Reversion: The Meeting (Steam, November 2019) (Simplified Chinese)", according to checksums for files "data.dcp", "Linux.dcp", "xlanguage_nz.dcp" and "xlanguage_zh_s.dcp".

File: engines/wintermute/detection_tables.h [hunk 1]
```diff
@@ -991,6 +1006,14 @@ static const WMEGameDescription gameDescriptions[] = {
 		WME_ENTRY2s("german.dcp", "bf4c2b8c26342342441a6d64934ab832", 107027865,
 					"data.dcp", "50de0beaa5ad621aa9f020df901d1e74", 1342214), Common::DE_DEU, ADGF_UNSTABLE, WME_1_9_1),
 
+	// Limbo of the Lost (English)
+	WME_WINENTRY("lotl", "",
```
Comment (lolbot-iichan) [inline engines/wintermute/detection_tables.h]: Is it a 2D or 2.5D game? I don't have it, but it looks 2.5D on youtube.
In this case we need "NOTE: This is a 2.5D game that is out of ScummVM scope" here.

Comment (aviloria) [reply engines/wintermute/detection_tables.h]: You're totally right.. I'll fix it

Comment (lolbot-iichan) [reply engines/wintermute/detection_tables.h]: The same applies to "wmedemo3d".

**Comment by aviloria** [comment]
About the chinese language conflict.. you're alse right.
I misunderstood what you mean because I focused on Windows versions when is a Linux problem.
I'm fixing it right now usign your first approach: adding a ZH_ANY flag for them.
As long as I test it I'll commit my chanegs

File: engines/wintermute/detection_tables.h (hunk 1 above)
Comment (aviloria) [reply engines/wintermute/detection_tables.h]: Niche catches! I overlooked them!

**Comment by lolbot-iichan** [comment]
@sev- Is it OK to add ZH_ANY language id? We have a game that has actual ZH_CNA, ZH_TWN translations and also ships with the same file set that formed ZH_CNA in previous game versions, when there was no separate Traditional/Simplified variants and was only one "Chinese".

Pro: We already have EN_ANY (when only one game version exist), why not?
Con: There is no such thing as generic Chinese. You either spell "魚" or "鱼".
Pro: There is no such thing as generic English as well. You either spell "color" or "colour".
Con: There are some games marked as ZH_CNA that does not have ZH_TWN. What about "(when only one game version exist)" part?

**Comment by aviloria** [comment]
@sev- Any comment about the Chinesse language?

**Comment by lolbot-iichan** [comment]
@sev- @bluegr @digitall 
Anyone?...

**Comment by digitall** [comment]
Can't speak for others, but am quite busy IRL currently.

**Comment by sev-** [comment]
This has to be rebased. I'll review it tomorrow.

**Comment by bluegr** [comment]
The changes look OK, nice work!

The commit that adds the new Chinese language could be cherry picked. The rest should be squashed.

**Comment by sev-** [comment]
Also, commits which change line endings need to be removed. And whitespace changes to the detection table as well. Currently, the patch is huge, but still useless: mostly those are pointless whitespace changes. Please, revert those parts.

**Comment by sev-** [comment]
@aviloria ZH_ANY is perfectly fine. Please fix up the unneeded changes, squash couple of commits and it is goot to merge.

**Comment by bluegr** [comment]
I've cherry picked commit 9fffd17a2435dcc5a7595d82bf504e9975b9a1ab and will squash the rest

**Comment by aviloria** [comment]
Sorry for the late... I was far from home on Chistmas days,
Thank you so much for the review and the needed changes,
Merry Christmas for all of you!

**Pull request merged**